commands/gist.go

```diff
@@ -0,0 +1,201 @@
+package commands
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/github/hub/github"
+	"github.com/github/hub/utils"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+)
+
+var cmdGist = &Command{
+	Run: gist,
+	Usage: `
+gist <GistID> [--no-headers] [<filename>]
+gist [--public] --file <file>
```

Comment:
When creating gists, I think this should be the preferred usage:

```
hub gist <file1> <file2...>
```

i.e. files are explicitly listed as arguments, without needing the `--file` flag. And if there are no files listed as arguments, stdin is used. What do you think?

Reply:
that only works if we drop the ability to fetch gists, which, as described above, defeats the purpose.

I'm OK if we want to say `--files <file1> <file2>` - That said in all my time using GH I've never ever seen anyone use more than one file in a Gist... I didn't even register it was possible until I started writing this... it's pastebin for output and code.